ceph_wrapper.py

```diff
@@ -0,0 +1,217 @@
+        except Exception as e:
+            raise e
+
+
+
+class RBD(CephBase):
+    def list_n(self):
+        try:
+            return self.rbd.list(self.ctx)
+        except Exception as e:
+            raise e
+          
+    def clone(self, p_name, p_snap, c_nm,\
+            p_ctx = None, c_ctx = None):
+        try:
+            if not p_ctx:
+                p_ctx = self.ctx
+            if not c_ctx:
+                c_ctx = self.ctx
+            self.rbd.clone(p_ctx, p_name, p_snap, c_ctx,\
+                    c_nm, features = 1 )
+            if c_nm in self.list_n():
+                self.c_img_list.append(c_nm)
+                return True
+            else:
+                raise CephException("post image creation check failed")
+        except Exception as e:
+            raise e
+    
+    def remove(self, iname, ctx = None):
+        if not ctx:
+            ctx = self.ctx
+        if iname in self.c_img_list:
+            self._remove(iname, ctx) 
+        else:
+            raise CephException("image not found in list,'\
+                    ' use the unsafe remove if you are sure")
+        if iname not in self.list_n():
+            self.c_img_list.remove(iname)
+            return True
+        else:
+            raise CephException("post image removal\
+                    ' check failed")
+
+    def _remove(self, iname, ctx = None):
+            try:
+                if not ctx:
+                    ctx = self.ctx
+                self.rbd.remove(ctx, iname)
+            except CephException as e:
+                raise e
+
+    def write(img_name, data, offset):
+        try:
+            if not img_name in self.img_dict:
+                raise CephException('Image not instantiated:'\
+                        ' Open the image first')
+            self.img_dict[img_name].write(data, offset)
+        except Exception as e:
+            raise e
+
+    def close_image(self, name):
+        if name in self.img_dict:
+            self.img_dict[name].close()
+            del(self.img_dict[name])
+        else:
+            raise CephException("{0} not found!".format(name))
+
+    def snap_image(self, img_name, name):
+        try:
+            if img_name not in self.img_dict:
+                raise CephException("Invalid image name"\
+                        " or image not instantiated")
+            self.img_dict[img_name].create_snap(name)
+            return True
+        except Exception as e:
+            raise e
+    
+    def list_snapshots(self, img_name):
+        try:
+            if img_name not in self.img_dict:
+                raise CephException("Invalid image name"\
+                        " or image not instantiated")
+            snap_list_iter = self.img_dict[img_name].list_snaps()
+            snap_list = [snap['name'] for snap in snap_list_iter]
+            return snap_list
+        except Exception as e:
+            raise e
+    
+    def remove_snapshots(self, img_name, name, debug = False):
```

Comment:
Maybe I'm missing something, but it doesn't seem like this method can remove more than one snapshot at a time. Because of this could the "s" in the method name be removed, so it's `remove_snapshot`? I know this is pretty minor, but it may be confusing for someone else using this method.